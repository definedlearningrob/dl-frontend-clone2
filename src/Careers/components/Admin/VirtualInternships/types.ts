import { TOpportunity as TBaseOpportunity } from '@dc/resources/types';

export const enum VirtualInternshipStatuses {
  DRAFT = 'DRAFT',
  PUBLISHED = 'PUBLISHED',
}

export type VirtualInternshipLessonAttributes = {
  lessonId: string;
  step: number;
};

export type FormValues = {
  badges: {
    id: string;
    name: string;
    imageUrl: string;
  }[];
  description: string;
  name: string;
  pathways: {
    label: string;
    value: string;
  }[];
  availableSpots: number;
  creditsOutcomes: string;
  image: Record<string, unknown>;
  tags: {
    label: string;
    value: string;
  }[];
  requiredExperiences: number;
  status: { value: VirtualInternshipStatuses; label: string };
  calendarLessons: Lesson[];
  experienceOpportunityLessons: Lesson[];
  postExperienceLessons: Lesson[];
  readinessSkillsLessons: Lesson[];
};

export type TPathway = Pick<TBaseOpportunity['pathways'][number], 'id' | 'name'>;

export type TOpportunity = Pick<
  TBaseOpportunity,
  | 'id'
  | 'name'
  | 'availableSpots'
  | 'creditsOutcomes'
  | 'description'
  | 'imageUrl'
  | 'opportunityType'
  | 'tags'
  | 'salaryInformation'
> & {
  pathways: TPathway[];
};

export type Lesson = {
  id: string;
  step: number;
  name: string;
  imageUrl: string;
  thumbnailUrl: string;
  type: string;
};

export type TVirtualInternship = {
  archivedAt: string | null;
  badges: {
    id: string;
    name: string;
    imageUrl: string;
  }[];
  id: string;
  opportunity: TOpportunity;
  requiredExperiences: number;
  status: VirtualInternshipStatuses;
  calendarLessons: Lesson[];
  experienceOpportunityLessons: Lesson[];
  postExperienceLessons: Lesson[];
  readinessSkillsLessons: Lesson[];
};
