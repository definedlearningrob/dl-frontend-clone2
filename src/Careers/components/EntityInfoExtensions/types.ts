export type TAdminEntityInfoExtensionField = {
  id: string;
  name: string;
};
