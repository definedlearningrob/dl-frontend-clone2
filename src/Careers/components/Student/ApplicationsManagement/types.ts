export type Counselor = {
  email: string;
  firstName: string;
  fullName: string;
  lastName: string;
  name: string;
  username: string;
  uuid: string;
};

export type AvailableTeacher = {
  email: string;
  fullName: string;
  name: string;
  username: string;
  uuid: string;
};

export type CounselorSelectOption = { label: string; value: string };
