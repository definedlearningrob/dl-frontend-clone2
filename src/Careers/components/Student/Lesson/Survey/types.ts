export type SURVEY_QUESTION_TYPE = 'single_choice' | 'multiple_choice';

type CareerReviewSurveyQuestionOption = {
  option: string;
  step: number;
};

type CareerReviewSurveyQuestion = {
  answer: string[];
  id: string;
  options: CareerReviewSurveyQuestionOption[];
  question: string;
  type: SURVEY_QUESTION_TYPE;
};

export type CareerReviewSurvey = {
  __typename: string;
  questions: CareerReviewSurveyQuestion[];
  performed: boolean;
  version: number;
};
