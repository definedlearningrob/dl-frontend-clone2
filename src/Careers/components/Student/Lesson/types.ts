import { TRubricResult } from '@dc/graphql/user/queries/studentCourseActivity';

import {
  TCheckInGroup as TBaseCheckInGroup,
  TCheckInQuestion as TBaseCheckInQuestion,
} from '@shared/components/CheckIns/types';

import { SURVEY_QUESTION_TYPE } from './Survey/types';

type TRubricHeading = {
  id: string;
  multiplier: number;
  name: string;
  uuid: string;
};

type TRubricCriteria = {
  id: string;
  rubricCriteriaLabelId: string;
  rubricHeadingId: string;
  text: string;
  uuid: string;
};

type TRubricCriteriaLabel = {
  displayName: string;
  id: string;
  score: number;
  uuid: string;
};

type TRubric = {
  id: string;
  uuid: string;
  name: string;
  description: string;
  headings: TRubricHeading[];
  criterias: TRubricCriteria[];
  pointsAvailable?: number;
  criteriaLabels: TRubricCriteriaLabel[];
  hasAlignedStatements?: boolean;
};

type TAssignmentSubmissionFile = {
  filename: string;
  googleWeblink?: string;
  id: string;
  source: string;
  url: string;
};

type TSubmissionGrade = {
  id: string;
  lastGradedBy: {
    uuid: string;
    firstName: string;
    lastName: string;
    fullName: string;
  };
  status: 'ACCEPTED' | 'NOT_ACCEPTED';
  createdAt: string;
  updatedAt: string;
};

type TSubmissionRubricGrade = {
  id: string;
  lastGradedBy: {
    uuid: string;
    firstName: string;
    lastName: string;
    fullName: string;
  };
  results: TRubricResult[];
  pointsAvailable: number;
  pointsScored: number;
  createdAt: string;
  updatedAt: string;
};

export type TAssignmentSubmission = {
  files: TAssignmentSubmissionFile[];
  grade: TSubmissionGrade | null;
  rubricGrade: TSubmissionRubricGrade | null;
  id: string;
  updatedAt: string;
  status: 'DRAFT' | 'SUBMITTED';
};

type TAttachmentFile = {
  filename: string;
  id: string;
  url: string;
};

type TExtensionField = {
  author: {
    email: string;
    firstName: string;
    lastName: string;
    username: string;
    uuid: string;
  };
  description: string;
  files: {
    filename: string;
    id: string;
    url: string;
  }[];
  id: string;
  imageUrl: string;
  links: {
    name: string;
    url: string;
  }[];
  name: string;
};

export type TAssignment = {
  id: string;
  name: string;
  description: string;
  displayName: string;
  step: number;
  rubrics: TRubric[];
  updatedAt: string;
  submission: TAssignmentSubmission | null;
  __typename: 'Assignment';
};

export type TAttachment = {
  description: string;
  displayName: string;
  files: TAttachmentFile[];
  id: string;
  step: number;
  __typename: 'Attachment';
};

export type TCareerReviewSurvey = {
  version: number;
  performed: boolean;
  questions: {
    answer: string[];
    id: string;
    options: {
      option: string;
      step: number;
    }[];
    question: string;
    type: SURVEY_QUESTION_TYPE;
  }[];
};

export type TExternalPresentation = {
  displayName: string;
  id: string;
  source: string;
  step: number;
  isExpandable: boolean;
  __typename: 'ExternalPresentation';
};

export type TResearchLink = {
  author: string;
  displayName: string;
  id: string;
  resourceLink: string;
  sourceName: string;
  step: number;
  __typename: 'ResearchLink';
};

export type TText = {
  content: string;
  displayName: string;
  id: string;
  step: number;
  __typename: 'Text';
};

export type TVideo = {
  description: string;
  displayName: string;
  filename: string;
  id: string;
  step: number;
  url: string;
  __typename: 'Video';
};

export type TVocabulary = {
  definition: string;
  id: string;
  step: number;
  term: string;
  __typename: 'Vocabulary';
};

export type TCheckInQuestion = Pick<
  TBaseCheckInQuestion,
  'id' | 'question' | 'step' | 'answer' | '__typename'
>;

export type TCheckInGroup = Omit<TBaseCheckInGroup, 'questions'> & {
  questions: TCheckInQuestion[];
};

export type TLesson = {
  id: string;
  imageUrl: string;
  name: string;
  assignments: TAssignment[];
  attachments: TAttachment[];
  careerReviewSurvey: TCareerReviewSurvey | null;
  checkInGroups: TCheckInGroup[];
  checkInQuestions: TCheckInQuestion[];
  description: {
    audience: string;
    goal: string;
    introduction: string;
    role: string;
    situation: string;
  };
  extensionFields: TExtensionField[];
  externalPresentations: TExternalPresentation[];
  hasPresentation: boolean;
  progress: {
    accepted: number;
    submitted: number;
    total: number;
  };
  researchLinks: TResearchLink[];
  step: number;
  texts: TText[];
  thumbnailUrl: string;
  type: string;
  videos: TVideo[];
  vocabularies: TVocabulary[];
};
