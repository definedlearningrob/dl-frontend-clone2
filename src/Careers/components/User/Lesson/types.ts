import { SURVEY_QUESTION_TYPE } from '@dc/components/Student/Lesson/Survey/types';

import {
  TCheckInGroup as TBaseCheckInGroup,
  TCheckInQuestion as TBaseCheckInQuestion,
} from '@shared/components/CheckIns/types';

type TRubricHeading = {
  id: string;
  multiplier: number;
  name: string;
  uuid: string;
};

type TRubricCriteria = {
  id: string;
  rubricCriteriaLabelId: string;
  rubricHeadingId: string;
  text: string;
  uuid: string;
};

type TRubricCriteriaLabel = {
  displayName: string;
  id: string;
  score: number;
  uuid: string;
};

type TRubric = {
  id: string;
  uuid: string;
  name: string;
  description: string;
  headings: TRubricHeading[];
  criterias: TRubricCriteria[];
  criteriaLabels: TRubricCriteriaLabel[];
};

type TAssignmentSubmissionFile = {
  filename: string;
  googleWeblink?: string;
  id: string;
  source: string;
  url: string;
};

type TSubmissionGrade = {
  id: string;
  lastGradedBy: {
    uuid: string;
    firstName: string;
    lastName: string;
    fullName: string;
  };
  status: 'ACCEPTED' | 'NOT_ACCEPTED';
  createdAt: string;
  updatedAt: string;
};

export type TAssignmentSubmission = {
  files: TAssignmentSubmissionFile[];
  grade: TSubmissionGrade | null;
  id: string;
  status: 'DRAFT' | 'SUBMITTED';
};

type TAttachmentFile = {
  filename: string;
  id: string;
  url: string;
};

type TExtensionField = {
  archivedAt: string | null;
  author: {
    email: string;
    firstName: string;
    lastName: string;
    username: string;
    uuid: string;
  };
  description: string;
  files: {
    filename: string;
    id: string;
    url: string;
  }[];
  id: string;
  imageUrl: string;
  links: {
    name: string;
    url: string;
  }[];
  name: string;
};

type TBadge = {
  archivedAt: string | null;
  id: string;
  imageUrl: string;
  name: string;
};

export type TAssignment = {
  archivedAt: string | null;
  assetName: string;
  badges: TBadge[];
  createdAt: string;
  id: string;
  name: string;
  description: string;
  displayName: string;
  step: number;
  rubrics: TRubric[];
  submission: TAssignmentSubmission | null;
  lessons: TLesson[];
  __typename: 'Assignment';
};

export type TAttachment = {
  archivedAt: string | null;
  createdAt: string;
  description: string;
  displayName: string;
  files: TAttachmentFile[];
  id: string;
  step: number;
  lessons: TLesson[];
  name: string;
  __typename: 'Attachment';
};

export type TCareerReviewSurvey = {
  questions: {
    answer: string[];
    id: string;
    options: {
      option: string;
      step: number;
    }[];
    question: string;
    type: SURVEY_QUESTION_TYPE;
  }[];
};

export type TExternalPresentation = {
  archivedAt: string | null;
  displayName: string;
  id: string;
  source: string;
  step: number;
  isExpandable: boolean;
  __typename: 'ExternalPresentation';
};

export type TResearchLink = {
  archivedAt: string | null;
  author: string;
  createdAt: string;
  displayName: string;
  id: string;
  resourceLink: string;
  sourceName: string;
  step: number;
  __typename: 'ResearchLink';
};

export type TText = {
  archivedAt: string | null;
  content: string;
  createdAt: string;
  displayName: string;
  id: string;
  step: number;
  __typename: 'Text';
};

export type TVideo = {
  archivedAt: string | null;
  createdAt: string;
  description: string;
  displayName: string;
  filename: string;
  id: string;
  step: number;
  url: string;
  __typename: 'Video';
};

export type TVocabulary = {
  archivedAt: string | null;
  createdAt: string;
  definition: string;
  id: string;
  step: number;
  term: string;
  __typename: 'Vocabulary';
};

export type TCheckInQuestion = Pick<
  TBaseCheckInQuestion,
  'id' | 'question' | 'step' | 'answer' | '__typename'
>;

export type TCheckInGroup = Omit<TBaseCheckInGroup, 'questions'> & {
  questions: TCheckInQuestion[];
};

export type TLesson = {
  archivedAt: string | null;
  id: string;
  imageUrl: string;
  name: string;
  assignments: TAssignment[];
  attachments: TAttachment[];
  badges: TBadge[];
  careerReviewSurvey: TCareerReviewSurvey | null;
  checkInGroups: TCheckInGroup[];
  checkInQuestions: TCheckInQuestion[];
  description: {
    audience: string;
    goal: string;
    introduction: string;
    role: string;
    situation: string;
  };
  extensionFields: TExtensionField[];
  externalPresentations: TExternalPresentation[];
  hasPresentation: boolean;
  researchLinks: TResearchLink[];
  step: number;
  texts: TText[];
  thumbnailUrl: string;
  type: string;
  videos: TVideo[];
  vocabularies: TVocabulary[];
};
