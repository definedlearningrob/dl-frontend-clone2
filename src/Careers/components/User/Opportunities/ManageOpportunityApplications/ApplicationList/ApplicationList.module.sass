@import '@shared/stylesheets/config.sass'

$grid-template-columns: 1fr 168px 168px
$grid-template-columns-xxxl: 1fr 192px 176px

.studentNameWrapper
  +flex-container(row, flex-start, center)
  column-gap: $m-xs

.table
  :global(.shared-table__column-1)
    padding-left: $m-base

    +media-breakpoint-up(xxxl)
      padding-left: $m-md

  :global(.shared-table__column-3)
    padding-right: $m-base

    +media-breakpoint-up(xxxl)
      padding-right: $m-md

  :global(.shared-table__head),
  :global(.shared-table__body)
    grid-template-columns: $grid-template-columns

    +media-breakpoint-up(xxxl)
      grid-template-columns: $grid-template-columns-xxxl

.table,
.tableWrapper
  height: 100% !important

.tableColumn
  +flex-container(row, false, center)
  background-color: inherit

.tableHead
  padding-top: $m-md
  color: $color-neutral-700

  @extend .tableColumn
  font-size: $font-size-xxs
  font-weight: $font-weight-medium
  line-height: $line-height-lg

  +media-breakpoint-up(xxxl)
    font-size: $font-size-xs

.status
  display: inline-block

.emptyApplicationListWrapper
  +flex-container(column, center)
  height: 100%
