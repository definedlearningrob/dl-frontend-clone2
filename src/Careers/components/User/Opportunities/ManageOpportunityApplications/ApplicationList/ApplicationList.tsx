import { useTranslation } from 'react-i18next';
import cx from 'classnames';
import { useParams } from 'react-router-dom';
import { useState } from 'react';

import SharedTable from '@dc/shared/Table/Table';
import { TOpportunityApplication } from '@dc/resources/types';
import { useOpportunityApplicationQuery } from '@dc/graphql/user/hooks/useOpportunityApplicationQuery';

import Avatar from '@shared/components/Avatar/Avatar';
import { formatDateTime, parseDate } from '@shared/utils/date';
import { Badge } from '@shared/components/Badge/Badge';
import useQueryParams from '@shared/hooks/useQueryParams';
import { Tooltip } from '@shared/components/Tooltip';
import { useBreakpointUp } from '@shared/hooks/useBreakpointUp';

import styles from './ApplicationList.module.sass';

type RowData = {
  opportunityId: string;
  id: string;
  student: {
    fullName: string;
    uuid: string;
  };
  appliedAt: string;
  status: string;
};

type Props = {
  studentNameFilter: string;
};

export const ApplicationList = ({ studentNameFilter }: Props) => {
  const { t } = useTranslation();
  const isFullHD = useBreakpointUp({ breakpoint: 'xxxl' });
  const { id } = useParams<{ id: string }>();
  const [currentPage, setCurrentPage] = useState(1);

  const { data, fetchMore } = useOpportunityApplicationQuery({
    id,
    filter: { studentFullNameCont: studentNameFilter },
  });

  const applicationsWithFallback = data?.opportunity.filteredApplications.nodes || [];

  const { params, updateQueryParams } = useQueryParams<{ applicationId: string }>();

  const headers = [
    { title: t('opportunityManageApplications.list.studentName') },
    { title: t('opportunityManageApplications.list.status') },
    { title: t('opportunityManageApplications.list.applicationDate') },
  ];

  const getSelectedRowIndex = () => {
    const applicationIndex = applicationsWithFallback.findIndex(
      (application: TOpportunityApplication) => application.id === params.applicationId
    );

    return applicationIndex === -1 ? 0 : applicationIndex;
  };

  const tableConstants = [
    {
      render: (rowData: TOpportunityApplication) => {
        const studentNameClassName = cx(styles.studentNameWrapper, {
          'text-white': rowData.id === params.applicationId,
        });

        return (
          <div className={studentNameClassName}>
            <Avatar
              className={styles.image}
              label={rowData.student.fullName}
              size={isFullHD ? '40' : '32'}
            />
            <div className='text-xs xxxl:text-sm font-medium'>{rowData.student.fullName}</div>
          </div>
        );
      },
    },
    {
      render: (rowData: RowData) => {
        let badgeType: 'primary' | 'secondary' | 'danger' = 'secondary';
        let badgeLabel: string = '';

        switch (rowData.status) {
          case 'PENDING':
            badgeType = 'secondary';
            badgeLabel = t('opportunityManageApplications.list.applicationStatus.pending');
            break;
          case 'ACCEPTED':
            badgeType = 'primary';
            badgeLabel = t('opportunityManageApplications.list.applicationStatus.accepted');
            break;
          case 'REJECTED':
            badgeType = 'danger';
            badgeLabel = t('opportunityManageApplications.list.applicationStatus.rejected');
        }

        return <Badge type={badgeType}>{badgeLabel}</Badge>;
      },
    },
    {
      render: (rowData: RowData) => {
        const dateClassName = cx('text-xs xxxl:text-sm', {
          'text-white': rowData.id === params.applicationId,
        });

        return (
          <Tooltip message={formatDateTime(rowData.appliedAt, { withTime: true })}>
            <span className={dateClassName}>{parseDate(rowData.appliedAt)}</span>
          </Tooltip>
        );
      },
    },
  ];

  const pagesCount = data?.opportunity.filteredApplications.pagesCount || 0;
  const hasNextPage = currentPage < pagesCount;

  const handleFetchMore = () => {
    const nextPage = currentPage + 1;

    fetchMore({
      variables: { filter: { studentFullNameCont: studentNameFilter }, page: nextPage },
    });

    setCurrentPage(nextPage);
  };

  return (
    <SharedTable tableClassname={styles.table} tableWrapperClassname={styles.tableWrapper}>
      <SharedTable.Head cols={headers} columnClassname={styles.tableHead} />
      <SharedTable.Body
        cols={tableConstants}
        columnClassname='flex items-center bg-inherit'
        data={applicationsWithFallback}
        fetchMore={handleFetchMore}
        hasNextPage={hasNextPage}
        rowClassname='cursor-pointer hover:bg-primary-200'
        selectedRowClassname='bg-primary-500 hover:!bg-primary-500'
        selectedRowIndex={getSelectedRowIndex()}
        onRowClick={(_, application) => updateQueryParams({ applicationId: application.id })}
      />
    </SharedTable>
  );
};
