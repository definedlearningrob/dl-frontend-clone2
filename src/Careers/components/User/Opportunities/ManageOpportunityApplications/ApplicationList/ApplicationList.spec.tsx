import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import { Route } from 'react-router-dom';
import { waitFor, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { OpportunityTypes } from '@graphql/dc/shared/types';

import { renderWithRouterAndReduxProvider } from '@dc/utils/test';
import { ApplicationList } from '@dc/components/User/Opportunities/ManageOpportunityApplications/ApplicationList/ApplicationList';
import {
  OPPORTUNITY_APPLICATIONS_QUERY,
  TOpportunityApplicationVariables,
  TOpportunityData,
} from '@dc/graphql/user/queries/opportunityApplications';
import { OPPORTUNITY_APPLICATION_STATUS } from '@dc/resources/enums';

const application = {
  id: '1',
  appliedAt: '2023-05-01',
  updatedAt: '2023-05-02',
  status: OPPORTUNITY_APPLICATION_STATUS.PENDING,
  answers: [{ id: '1', opportunityQuestionId: '1', answer: 'To sleep better!', step: 1 }],
  lastChangedBy: {
    uuid: '123',
    name: 'Teacher Name',
  },
  student: {
    uuid: '123',
    fullName: 'Applicant Student 1',
    schoolClasses: [{ uuid: 'class-uuid', name: 'Class name' }],
  },
};

const opportunity = {
  hasPendingApplications: true,
  questions: [
    { id: '1', question: 'Why you want to test this?', answer: 'To sleep better!', step: 1 },
  ],
  opportunityType: OpportunityTypes.APPRENTICESHIP,
  automaticAcceptance: false,
  id: '1',
  uuid: '123',
  applications: { nodes: [application], pagesCount: 1, nodesCount: 1 },
  filteredApplications: {
    nodes: [
      application,
      {
        ...application,
        id: '2',
        status: OPPORTUNITY_APPLICATION_STATUS.ACCEPTED,
        student: {
          uuid: '111',
          fullName: 'Applicant Student 2',
          schoolClasses: [{ uuid: 'class-uuid', name: 'Class name' }],
        },
      },
      {
        ...application,
        id: '3',
        status: OPPORTUNITY_APPLICATION_STATUS.REJECTED,
        student: {
          uuid: '222',
          fullName: 'Applicant Student 3',
          schoolClasses: [{ uuid: 'class-uuid', name: 'Class name' }],
        },
      },
      {
        ...application,
        id: '4',
        status: OPPORTUNITY_APPLICATION_STATUS.STARTED,
        student: {
          uuid: '333',
          fullName: 'Applicant Student 4',
          schoolClasses: [{ uuid: 'class-uuid', name: 'Class name' }],
        },
      },
    ],
    pagesCount: 1,
    nodesCount: 1,
  },
  name: 'Opportunity name',
};

const applicationsListSpy = jest.fn();

const opportunityApplicationsMock: MockedResponse<
  TOpportunityData,
  TOpportunityApplicationVariables
> = {
  request: {
    query: OPPORTUNITY_APPLICATIONS_QUERY,
    variables: {
      filter: { studentFullNameCont: '' },
      perPage: 1000,
      page: undefined,
      id: '1',
    },
  },
  result() {
    applicationsListSpy();

    return { data: { opportunity } };
  },
};
describe('ApplicationList', () => {
  it('should render correctly', async () => {
    const { container } = renderWithRouterAndReduxProvider(
      <Route path='/opportunities/:id/manage-applications'>
        <MockedProvider mocks={[opportunityApplicationsMock]}>
          <ApplicationList studentNameFilter='' />
        </MockedProvider>
      </Route>,
      { route: '/opportunities/1/manage-applications' }
    );

    await waitFor(() => {
      expect(applicationsListSpy).toHaveBeenCalledTimes(1);
    });

    expect(container).toMatchSnapshot();
  });

  it('should select application on row click', async () => {
    const { history } = renderWithRouterAndReduxProvider(
      <Route path='/opportunities/:id/manage-applications'>
        <MockedProvider mocks={[opportunityApplicationsMock]}>
          <ApplicationList studentNameFilter='' />
        </MockedProvider>
      </Route>,
      { route: '/opportunities/1/manage-applications' }
    );

    userEvent.click(await screen.findByText('Applicant Student 1'));
    expect(history.location.search).toContain('?applicationId=1');

    userEvent.click(await screen.findByText('Applicant Student 2'));
    expect(history.location.search).toContain('?applicationId=2');
  });
});
