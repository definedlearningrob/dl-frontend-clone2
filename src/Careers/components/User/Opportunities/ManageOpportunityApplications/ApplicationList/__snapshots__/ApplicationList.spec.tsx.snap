// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationList should render correctly 1`] = `
<div>
  <div
    class="shared-table tableWrapper"
  >
    <table
      class="shared-table__table table"
    >
      <thead
        class="shared-table__head shared-table__head-cols-3"
      >
        <tr
          class="shared-table__row shared-table__row-0"
        >
          <th
            class="shared-table__column -header-cell shared-table__column-1 tableHead"
          >
            Student name
          </th>
          <th
            class="shared-table__column -header-cell shared-table__column-2 tableHead"
          >
            Status
          </th>
          <th
            class="shared-table__column -header-cell shared-table__column-3 tableHead"
          >
            Application date
          </th>
        </tr>
      </thead>
      <tbody
        class="shared-table__body shared-table__body-cols-3"
      >
        <tr
          class="shared-table__row shared-table__row-1 cursor-pointer hover:bg-primary-200 bg-primary-500 hover:!bg-primary-500"
          data-testid="list-item"
        >
          <td
            class="shared-table__column shared-table__column-1 flex items-center bg-inherit"
            data-testid="shared-table-column"
          >
            <div
              class="studentNameWrapper"
            >
              <div
                class="avatar image"
                data-testid="avatar-image"
              >
                <div
                  class="avatar__image sb-avatar"
                  style="display: inline-block; vertical-align: middle; width: 32px; height: 32px; border-radius: 100%; font-family: Helvetica, Arial, sans-serif;"
                >
                  <div
                    class="avatar__image sb-avatar__text"
                    style="width: 32px; height: 32px; line-height: initial; text-align: center; color: rgb(116, 122, 136); background: rgb(240, 242, 247); border-radius: 100%;"
                    title="Applicant Student 1"
                  >
                    <div
                      style="display: table; table-layout: fixed; width: 100%; height: 100%;"
                    >
                      <span
                        style="display: table-cell; vertical-align: middle; font-size: 100%; white-space: nowrap;"
                      >
                        <span />
                      </span>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="text-xs xxxl:text-sm font-medium"
              >
                Applicant Student 1
              </div>
            </div>
          </td>
          <td
            class="shared-table__column shared-table__column-2 flex items-center bg-inherit"
            data-testid="shared-table-column"
          >
            <div
              class="font-medium leading-lg flex items-center bg-secondary-200 text-secondary-500 rounded-sm p-xxs text-xs gap-xxs"
            >
              Pending
            </div>
          </td>
          <td
            class="shared-table__column shared-table__column-3 flex items-center bg-inherit"
            data-testid="shared-table-column"
          >
            <span
              data-state="closed"
            >
              <span
                class="text-xs xxxl:text-sm"
              >
                May 1, 2023
              </span>
            </span>
          </td>
        </tr>
        <tr
          class="shared-table__row shared-table__row-2 cursor-pointer hover:bg-primary-200"
          data-testid="list-item"
        >
          <td
            class="shared-table__column shared-table__column-1 flex items-center bg-inherit"
            data-testid="shared-table-column"
          >
            <div
              class="studentNameWrapper"
            >
              <div
                class="avatar image"
                data-testid="avatar-image"
              >
                <div
                  class="avatar__image sb-avatar"
                  style="display: inline-block; vertical-align: middle; width: 32px; height: 32px; border-radius: 100%; font-family: Helvetica, Arial, sans-serif;"
                >
                  <div
                    class="avatar__image sb-avatar__text"
                    style="width: 32px; height: 32px; line-height: initial; text-align: center; color: rgb(116, 122, 136); background: rgb(240, 242, 247); border-radius: 100%;"
                    title="Applicant Student 2"
                  >
                    <div
                      style="display: table; table-layout: fixed; width: 100%; height: 100%;"
                    >
                      <span
                        style="display: table-cell; vertical-align: middle; font-size: 100%; white-space: nowrap;"
                      >
                        <span />
                      </span>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="text-xs xxxl:text-sm font-medium"
              >
                Applicant Student 2
              </div>
            </div>
          </td>
          <td
            class="shared-table__column shared-table__column-2 flex items-center bg-inherit"
            data-testid="shared-table-column"
          >
            <div
              class="font-medium leading-lg flex items-center bg-primary-200 text-primary-500 rounded-sm p-xxs text-xs gap-xxs"
            >
              Accepted
            </div>
          </td>
          <td
            class="shared-table__column shared-table__column-3 flex items-center bg-inherit"
            data-testid="shared-table-column"
          >
            <span
              data-state="closed"
            >
              <span
                class="text-xs xxxl:text-sm"
              >
                May 1, 2023
              </span>
            </span>
          </td>
        </tr>
        <tr
          class="shared-table__row shared-table__row-3 cursor-pointer hover:bg-primary-200"
          data-testid="list-item"
        >
          <td
            class="shared-table__column shared-table__column-1 flex items-center bg-inherit"
            data-testid="shared-table-column"
          >
            <div
              class="studentNameWrapper"
            >
              <div
                class="avatar image"
                data-testid="avatar-image"
              >
                <div
                  class="avatar__image sb-avatar"
                  style="display: inline-block; vertical-align: middle; width: 32px; height: 32px; border-radius: 100%; font-family: Helvetica, Arial, sans-serif;"
                >
                  <div
                    class="avatar__image sb-avatar__text"
                    style="width: 32px; height: 32px; line-height: initial; text-align: center; color: rgb(116, 122, 136); background: rgb(240, 242, 247); border-radius: 100%;"
                    title="Applicant Student 3"
                  >
                    <div
                      style="display: table; table-layout: fixed; width: 100%; height: 100%;"
                    >
                      <span
                        style="display: table-cell; vertical-align: middle; font-size: 100%; white-space: nowrap;"
                      >
                        <span />
                      </span>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="text-xs xxxl:text-sm font-medium"
              >
                Applicant Student 3
              </div>
            </div>
          </td>
          <td
            class="shared-table__column shared-table__column-2 flex items-center bg-inherit"
            data-testid="shared-table-column"
          >
            <div
              class="font-medium leading-lg flex items-center bg-danger-100 text-danger-500 rounded-sm p-xxs text-xs gap-xxs"
            >
              Rejected
            </div>
          </td>
          <td
            class="shared-table__column shared-table__column-3 flex items-center bg-inherit"
            data-testid="shared-table-column"
          >
            <span
              data-state="closed"
            >
              <span
                class="text-xs xxxl:text-sm"
              >
                May 1, 2023
              </span>
            </span>
          </td>
        </tr>
        <tr
          class="shared-table__row shared-table__row-4 cursor-pointer hover:bg-primary-200"
          data-testid="list-item"
        >
          <td
            class="shared-table__column shared-table__column-1 flex items-center bg-inherit"
            data-testid="shared-table-column"
          >
            <div
              class="studentNameWrapper"
            >
              <div
                class="avatar image"
                data-testid="avatar-image"
              >
                <div
                  class="avatar__image sb-avatar"
                  style="display: inline-block; vertical-align: middle; width: 32px; height: 32px; border-radius: 100%; font-family: Helvetica, Arial, sans-serif;"
                >
                  <div
                    class="avatar__image sb-avatar__text"
                    style="width: 32px; height: 32px; line-height: initial; text-align: center; color: rgb(116, 122, 136); background: rgb(240, 242, 247); border-radius: 100%;"
                    title="Applicant Student 4"
                  >
                    <div
                      style="display: table; table-layout: fixed; width: 100%; height: 100%;"
                    >
                      <span
                        style="display: table-cell; vertical-align: middle; font-size: 100%; white-space: nowrap;"
                      >
                        <span />
                      </span>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="text-xs xxxl:text-sm font-medium"
              >
                Applicant Student 4
              </div>
            </div>
          </td>
          <td
            class="shared-table__column shared-table__column-2 flex items-center bg-inherit"
            data-testid="shared-table-column"
          >
            <div
              class="font-medium leading-lg flex items-center bg-secondary-200 text-secondary-500 rounded-sm p-xxs text-xs gap-xxs"
            >
              
            </div>
          </td>
          <td
            class="shared-table__column shared-table__column-3 flex items-center bg-inherit"
            data-testid="shared-table-column"
          >
            <span
              data-state="closed"
            >
              <span
                class="text-xs xxxl:text-sm"
              >
                May 1, 2023
              </span>
            </span>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
